src/networking/systems.rs

```diff
@@ -0,0 +1,240 @@
+use std::{collections::HashMap};
+
+use bevy::prelude::{Commands, Entity, EventReader, Events, ResMut};
+use quinn::{crypto::rustls::TlsSession, generic::{RecvStream, SendStream}};
+use tokio::{stream::StreamExt, sync::mpsc::UnboundedReceiver, sync::mpsc::UnboundedSender, sync::mpsc::unbounded_channel};
+use tracing::{error, info};
+
+use super::{components::Connection, events::ReceiveEvent, events::SendEvent, id::ConnectionId, id::StreamId, packets::Packet, streams::BoundedPlanetRecvStream, streams::BoundedPlanetSendStream};
+
+// Internal state of the network session system
+pub struct SessionEventListenerState {
+    // Map from ConnectionID => StreamId => StreamSender
+    pub stream_senders: HashMap<ConnectionId, HashMap<StreamId, UnboundedSender<Packet>>>,
+
+    // MPSC which receives all events from the network
+    pub event_receiver: UnboundedReceiver<ReceiveEvent>,
+
+    // Event reader used for pulling send events and publishing them to the network
+    pub send_event_reader: EventReader<SendEvent>,
+}
+
+// ECS resources containing a map of active network connections
```

Comment:
```suggestion
/// ECS resources containing a map of active network connections
```
Might as well turn this into a doc comment since you already did the work to explain it